packages/core/src/lib/toast/__tests__/toast-banner.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { describe, it, expect, vi } from 'vitest';
+import { fireEvent, render, screen } from '@testing-library/svelte';
+
+import Toast from './toast-banner.spec.svelte';
+import { cxTestArguments, cxTestResults } from '$lib/__tests__/cx-test';
+import { ToastVariant } from '../variants';
+
+describe('Toast', () => {
+  it('Renders toast element with appropriate message, and action text', () => {
+    const { container } = render(Toast, {
+      variant: ToastVariant.Success,
+    });
+
+    expect(screen.getByText('This is the message.')).toBeVisible();
+    expect(screen.getByText('This is the action button.')).toBeVisible();
+    expect(container.querySelector('.text-success-dark')).toBeVisible();
+    expect(screen.getByRole('button')).toBeVisible();
```

Comment:
- Use data-testid for testing slots, as described above
- Try to always separate your element queries from you expectations for readability
- Try to always use names when you're querying buttons, especially when they're icon buttons, which need extra a11y care
- Almost never use `container` and almost never use `querySelector`
    - Needing to resort to these is usually a sign that your HTML isn't properly accessible and/or semantic
    - In this case, I would say that the `Icon` needs to be "visible" to the a11y tree as an image with a label of "success," because it conveys meaning for the toast message that isn't available elsewhere

```suggestion
    const message = screen.getByTestId('message')
    const action = screen.getByTestId('action')
    const icon = screen.getByRole('img', { name: /success/iu })
    const dismissButton = screen.getByRole('button', { name: /dismiss/iu })
    
    expect(message).toBeVisible();
    expect(action).toBeVisible();
    expect(icon).toHaveClass('text-success-dark');
    expect(dismissButton).toBeVisible();
```